resource/reapi/bindings/go/src/test/CMakeLists.txt

```
@@ -3,15 +3,23 @@ set(SRCS main.go)
 set(CGO_CFLAGS "-I${CMAKE_BINARY_DIR}/resource/reapi/bindings/c")
 
 # This is currently passed but not used because when passed into add_custom_command the spaces are escaped
-set(CGO_LIBRARY_FLAGS "-Wl,-R ${CMAKE_BINARY_DIR}/resource -L${CMAKE_BINARY_DIR}/resource/reapi/bindings -L${CMAKE_BINARY_DIR}/resource/libjobspec -ljobspec_conv -lreapi_cli -L${CMAKE_BINARY_DIR}/resource -lresource -L${CMAKE_BINARY_DIR}/resource/planner/c -lplanner -L${CMAKE_BINARY_DIR}/resource/planner/c++ -lplanner_cxx -lflux-idset -lstdc++ -lczmq -ljansson -lhwloc -lboost_system -lflux-hostlist -lboost_graph -lyaml-cpp")
+set(CGO_LIBRARY_FLAGS "-Wl,-R \
+    -lflux-idset \
+    -lstdc++ \
+    -lczmq \
+    -ljansson \
+    -lhwloc \
+    -lboost_system \
+    -lflux-hostlist \
+    -lboost_graph \
+    -lyaml-cpp \
+    -L${CMAKE_BINARY_DIR}/resource/reapi/bindings -lreapi_cli")
```

This is stylistic, but do you strongly prefer having this as a quoted string rather than unquoted here?  The extra noise of having to have a `\` on the end of each line is rather unfortunate.

---

I think it looks nice! It's much easier to read than the previous lump.

---

I agree, but I think we're hitting a mismatch of expectations on cmake syntax.  This following `set` command should build the same string as the one currently in the PR, and is slightly more idiomatic:

```suggestion
set(CGO_LIBRARY_FLAGS -Wl,-R
    -lflux-idset
    -lstdc++
    -lczmq
    -ljansson
    -lhwloc
    -lboost_system
    -lflux-hostlist
    -lboost_graph
    -lyaml-cpp
    -L${CMAKE_BINARY_DIR}/resource/reapi/bindings -lreapi_cli)
```

---

@trws for some reason neither your comment nor @vsoch displayed when I was responding. I tried the more cmake-idiomatic `set` command you suggested as well as a few others I devised, and couldn't get them to configure or link.

---

Ah, I made a mistake in the above example.  That actually gets stored as `"-Wl,-R;-flux-idset;-lstdc++"` etc., as a list, so it needs to be _unquoted_ on the usage to separate the list into args in the usage.

Alternately, this is a cmake-style multiline-string, it will include newlines in the string, but usually that's not really an issue as long as it isn't spliced into a string or similar.
```suggestion
set(CGO_LIBRARY_FLAGS [[ -Wl,-R
    -lflux-idset
    -lstdc++
    -lczmq
    -ljansson
    -lhwloc
    -lboost_system
    -lflux-hostlist
    -lboost_graph
    -lyaml-cpp
    -L${CMAKE_BINARY_DIR}/resource/reapi/bindings -lreapi_cli
    ]])
```

---

Unfortunately, the bracket argument method results in 
```bash
resource/reapi/bindings/go/src/test/CMakeFiles/main.dir/build.make:72: *** missing separator.  Stop.
```
The other issue is that bracket arguments don't support variable references (needed for `CMAKE_BINARY_DIR`): https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#bracket-argument

---

Fair enough. If you don't want to do the list expansion on the usage then leave as is. 👍

---

Ah, I got it! The list expansion style does work, but GolangSimple.cmake also needs to be changed to get rid of the replacement of spaces with `;`.

---

oops, that is definitely my bad :grimacing: :laughing: